### ISSUE TITLE
` curl -sfL https://get.k3s.io | sh -` did not work as expected

### ISSUE THREAD

**Issue by owen-carter** [rancher/k3s#361]
**Describe the bug**
`curl -sfL https://get.k3s.io | sh -` did not started the k3s service when I execute it

**To Reproduce**
1 .   curl -sfL https://get.k3s.io | sh -

**Expected behavior**
run as Expected

**Screenshots**
```sh
[root@iZm5e4rdade76nf13sbqfpZ ~]# journalctl -xe
-- Documentation: http://www.freedesktop.org/wiki/Software/systemd/multiseat
--
-- A new session with the ID 1 has been created for the user root.
--
-- The leading process of the session is 3235.
4月 17 08:21:53 iZm5e4rdade76nf13sbqfpZ sshd[3235]: pam_unix(sshd:session): session opened for user root by (uid=0)
4月 17 08:22:01 iZm5e4rdade76nf13sbqfpZ kernel: random: crng init done
4月 17 08:22:12 iZm5e4rdade76nf13sbqfpZ polkitd[1608]: Registered Authentication Agent for unix-process:3255:6891 (system bus name :1.17 [/usr/bin/pkttyagent --notify-fd 5 --fallback], object path /org/freedesktop/PolicyKit1/Authentica
4月 17 08:22:12 iZm5e4rdade76nf13sbqfpZ systemd[1]: Starting Lightweight Kubernetes...
-- Subject: Unit k3s.service has begun start-up
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit k3s.service has begun starting up.
4月 17 08:22:12 iZm5e4rdade76nf13sbqfpZ systemd[1]: k3s.service start-pre operation timed out. Terminating.
4月 17 08:22:12 iZm5e4rdade76nf13sbqfpZ systemd[1]: Failed to start Lightweight Kubernetes.
-- Subject: Unit k3s.service has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit k3s.service has failed.
--
-- The result is failed.
4月 17 08:22:12 iZm5e4rdade76nf13sbqfpZ systemd[1]: Unit k3s.service entered failed state.
4月 17 08:22:12 iZm5e4rdade76nf13sbqfpZ systemd[1]: k3s.service failed.
4月 17 08:22:12 iZm5e4rdade76nf13sbqfpZ polkitd[1608]: Unregistered Authentication Agent for unix-process:3255:6891 (system bus name :1.17, object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale zh_CN.UTF-8) (disconnected
4月 17 08:22:21 iZm5e4rdade76nf13sbqfpZ polkitd[1608]: Registered Authentication Agent for unix-process:3264:7781 (system bus name :1.18 [/usr/bin/pkttyagent --notify-fd 5 --fallback], object path /org/freedesktop/PolicyKit1/Authentica
4月 17 08:22:21 iZm5e4rdade76nf13sbqfpZ systemd[1]: Reloading.
4月 17 08:22:21 iZm5e4rdade76nf13sbqfpZ polkitd[1608]: Unregistered Authentication Agent for unix-process:3264:7781 (system bus name :1.18, object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale zh_CN.UTF-8) (disconnected
4月 17 08:22:23 iZm5e4rdade76nf13sbqfpZ polkitd[1608]: Registered Authentication Agent for unix-process:3281:7900 (system bus name :1.19 [/usr/bin/pkttyagent --notify-fd 5 --fallback], object path /org/freedesktop/PolicyKit1/Authentica
4月 17 08:22:23 iZm5e4rdade76nf13sbqfpZ systemd[1]: Starting Lightweight Kubernetes...
-- Subject: Unit k3s.service has begun start-up
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit k3s.service has begun starting up.
4月 17 08:22:23 iZm5e4rdade76nf13sbqfpZ systemd[1]: k3s.service start-pre operation timed out. Terminating.
4月 17 08:22:23 iZm5e4rdade76nf13sbqfpZ systemd[1]: Failed to start Lightweight Kubernetes.
-- Subject: Unit k3s.service has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit k3s.service has failed.
--
-- The result is failed.
4月 17 08:22:23 iZm5e4rdade76nf13sbqfpZ systemd[1]: Unit k3s.service entered failed state.
4月 17 08:22:23 iZm5e4rdade76nf13sbqfpZ systemd[1]: k3s.service failed.
4月 17 08:22:23 iZm5e4rdade76nf13sbqfpZ polkitd[1608]: Unregistered Authentication Agent for unix-process:3281:7900 (system bus name :1.19, object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale zh_CN.UTF-8) (disconnected
lines 2791-2838/2838 (END).
```

**Additional context**
```sh
[root@iZm5e4rdade76nf13sbqfpZ ~]# curl -sfL https://get.k3s.io | sh -
[INFO]  Finding latest release
[INFO]  Using v0.4.0 as release
[INFO]  Downloading hash https://github.com/rancher/k3s/releases/download/v0.4.0/sha256sum-amd64.txt
[INFO]  Skipping binary downloaded, installed k3s matches hash
[INFO]  Creating uninstall script /usr/local/bin/k3s-uninstall.sh
[INFO]  systemd: Creating environment file /etc/systemd/system/k3s.service.env
[INFO]  systemd: Creating service file /etc/systemd/system/k3s.service
[INFO]  systemd: Enabling k3s unit
[INFO]  systemd: Starting k3s
Job for k3s.service failed because a timeout was exceeded. See "systemctl status k3s.service" and "journalctl -xe" for details.
```

**Comment by erikwilson** [comment]
Related rancher/k3s/issues/360, probably caused by using `TimeoutStartSec=infinity` with an older version of systemd (https://stackoverflow.com/questions/42921055/changing-systemd-service-timeoutsec-value-to-infinity-has-no-effect)

**Comment by erikwilson** [comment]
@owen-carter could you also please provide the systemd version from `systemctl --version`?

**Comment by erikwilson** [comment]
Closing this is as fixed with PR #365, was tested in #360. If you think this is still an issue feel free to re-open with more information.
